src/syscheckd/db/include/db_file_item.hpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Wazuh Syscheckd
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * September 23, 2021.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _DBFILEITEM_HPP
+#define _DBFILEITEM_HPP
+#include "shared.h"
+#include "db_item.hpp"
+
+// Define EXPORTED for any platform
+#ifdef _WIN32
+#ifdef WIN_EXPORT
+#define EXPORTED __declspec(dllexport)
+#else
+#define EXPORTED __declspec(dllimport)
+#endif
+#elif __GNUC__ >= 4
+#define EXPORTED __attribute__((visibility("default")))
+#else
+#define EXPORTED
+#endif
+
+class EXPORTED FileItem : public DBItem {
+public:
+    FileItem(fim_entry*);
+    ~FileItem();
+    fim_entry* toFimEntry();
+    nlohmann::json* toJSON();
+
+private:
+    int                     m_options;
+    unsigned int            m_time;
+    unsigned int            m_size;
+    unsigned long           m_dev;
+    unsigned long int       m_inode;
+    std::string             m_attributes;
+    std::string             m_gid;
```

Comment:
The GID is an integer instead of a string.
```suggestion
    unsigned int            m_gid;
```

Reply:
```
typedef struct fim_file_data {
    // Checksum attributes
    unsigned int size;
    char * perm;
    char * attributes;
    char * uid;
    char * gid;
    char * user_name;
    char * group_name;
    unsigned int mtime;
    unsigned long int inode;
    os_md5 hash_md5;
    os_sha1 hash_sha1;
    os_sha256 hash_sha256;

    // Options
    fim_event_mode mode;
    time_t last_event;
    unsigned long int dev;
    unsigned int scanned;
    int options;
    os_sha1 checksum;
} fim_file_data; 
```

uid and gid are char* in fim_file_data struct so I convert this parameters in string